src/sources/util/http_client.rs

```diff
@@ -172,12 +174,23 @@ pub(crate) async fn call<
                                 start,
                                 end: Instant::now()
                             });
-                            context.on_response(&url, &header, &body).map(|events| {
+                            context.on_response(&url, &header, &body).map(|mut events| {
+                                let byte_size = if events.is_empty() {
+                                    0
+                                } else {
+                                    events.estimated_json_encoded_size_of()
+                                };
```

Comment:
Could the check for empty events and return zero, be moved into `estimated_json_encoded_size_of()` ?

It seems like that should be the case for all usages of that?

Reply:
I added a comment about why this is in place.